code/game/objects/items/robot/robot_parts.dm

```diff
@@ -199,6 +199,13 @@
 			if(!user.unEquip(W))
 				return
 
+			var/syndiemmi = 0
+			if(istype(W, /obj/item/device/mmi/syndie))

```

Comment:
add a variable to the base ```/obj/item/device/mmi``` item, ```var/syndiemmi``` and then just check for it here instead of typechecking here.